packages/governance/src/contractHelper.js

```diff
@@ -4,14 +4,30 @@ import { Far } from '@endo/marshal';
 import { makeStoredPublisherKit } from '@agoric/notifier';
 import { getMethodNames, objectMap } from '@agoric/internal';
 import { ignoreContext } from '@agoric/vat-data';
-import { keyEQ } from '@agoric/store';
+import { keyEQ, M } from '@agoric/store';
+import { AmountShape, BrandShape } from '@agoric/ertp';
 import { assertElectorateMatches } from './contractGovernance/paramManager.js';
 import { makeParamManagerFromTerms } from './contractGovernance/typedParamManager.js';
 
 const { details: X, quote: q } = assert;
 
 export const GOVERNANCE_STORAGE_KEY = 'governance';
 
+const extendPublicAPI = {
+  getSubscription: M.call().returns(M.remotable('StoredSubscription')),
+  getContractGovernor: M.call().returns(M.remotable('Instance')),
+  getGovernedParams: M.call().returns(M.record()),
+  getAmount: M.call().returns(AmountShape),
+  getBrand: M.call().returns(BrandShape),
+  getInstance: M.call().returns(M.remotable('Instance')),
+  getInstallation: M.call().returns(M.remotable('Installation')),
+  getInvitationAmount: M.call().returns(M.promise()),
```

Comment:
It's an Amount.  Can we embed the type inside a `Promise`?

Reply:
You can currently do `M.eref(AmountShape)`. But I'm thinking of removing `M.eref` because it looks like it is checking something stronger than it is. It checks that the specimen is either a promise or an amount. But if it is a promise, it does *not* check that the promise fulfills to an amount, because it cannot check that synchronously. We'd have the same problem with `M.promise(AmountShape)` except that it would never be stronger than "is it a promise". I omitted having such a parameter to avoid implying that more is being enforced than it is.

That said, we already live with this problem with `M.remotable`. Perhaps if the `M.promise` argument were just a string to be used in error diagnostics it would be clearer? I don't know. It makes me quesy in a way that `M.remotable(string)` doesn't. Opinions?

Reply:
Yeah, after a bit I realized that the argument to `M.promise()` can't be more than a hint for reporting and that's probably why you omitted it. 

If it was a string whose name was 'typeDescription' or 'nameForErrorReporting', would that make you more comfortable? My IDE shows me the names of parameters when it's hinting at completions.

How about if it were `M.call().returns(M.promise().hint('Invitation'))`?

Reply:
I would like exactly the same debugging support as for `remotable`. I'm not sure it's worth the additional `hint` usage. Particularly if it's a string, it's clear that's it's not actionable.

Reply:
Yeah, I think the uniformity with `M.remotable` is persuasive. I will suppress my queasiness ;)

Reply:
+1 to indicating what's expected inside the promise, even if it's not verified. This at least serves as documentation for the developer writing and debugging.

It could also be interpreted by software to create the function signature for `getInvitationAmount`, which needs to say what the function is supposed to do. Eventually it could be fed into the remote call handling to provide an automated check at runtime.

Reply:
I agree, but it can wait until those patterns have been enhanced with the debug info fields.